packages/edit-site/src/components/global-styles/use-global-styles-output.js

```diff
@@ -892,12 +1005,19 @@ export function useGlobalStylesOutput() {
 		return !! getSettings().disableLayoutStyles;
 	} );
 
+	const getBlockStyles = useSelect( ( select ) => {
+		return select( blocksStore ).getBlockStyles;
+	}, [] );
+
 	return useMemo( () => {
 		if ( ! mergedConfig?.styles || ! mergedConfig?.settings ) {
 			return [];
 		}
 		mergedConfig = updateConfigWithSeparator( mergedConfig );
-		const blockSelectors = getBlockSelectors( getBlockTypes() );
+		const blockSelectors = getBlockSelectors(
+			getBlockTypes(),
+			getBlockStyles
+		);
```

Comment:
Not a blocker, but in a code quality follow-up, would it be worth switching `getBlockSelectors` over to be a hook instead so we don't need to pass it the `getBlockStyles` function?